_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,91 @@
+The Blue Yonder Core Platform team implemented several work-arounds to allow for graceful downscaling, but this has not yet been addressed in a generic way in the distributed scheduler.
+ 
+ 
+During the working sessions, we had a number of fruitful conversations about stability with  Dask developers and users facing similar issues.
+One idea which may help mitigate the above issues is _replication of some task results_. The idea is the following: if a worker which is holding a task result is shut down, but the task result is replicated, then we do not need to re-compute parts of the graph again.
+One could think about replicating task results for tasks that qualify some criteria such as long running time or small task size.
+
+### Performance and graph optimization
+While monitoring the execution of one of our Dask production pipelines.
+The amount of memory consumed by the workers was significantly higher than one would expect in an ideal scenario where tasks are executed in a way which minimizes memory usage of the workers.
+When investigating the Dask.optimization module, we saw that code to optimize memory usage was already there. In practice, however, the graphs are not executed in such an order because of other constraints during execution.
+A Dask user at the workshop facing this issue told us that they worked around this by injecting dummy dependencies into Dask graphs. These dependencies acted as "choke-holds" for certain types of tasks, in order to improve the memory usage during execution.
+
+This lack of optimization referred to as memory back-pressure is currently not a priority for Blue Yonder, as increasing worker memory served as a work-around. There still may be value in exploring this further in the future.
+
+## Looking back
+
+### Results from working sessions
+Lucas had the pleasure to work with [John Lee](https://github.com/jlee3) on writing down a benchmark to enable [work stealing for tasks with restrictions](https://github.com/Dask/distributed/pull/3069), as well as with [Tom Augspurger](https://github.com/TomAugspurger) on a Dask.dataframe bug involving categoricals (PR still pending).
+
+However, arguably the most useful work we did was the implementation of a semaphore in Dask.distributed.
+We need to be able to rate-limit the access of the computation clusters to certain resources such as production databases. Migrating workflows which were still heavily dependent on accessing the database to the distributed scheduler was not possible without a semaphore implementation.
+By working on the semaphore during the workshop, we unblocked a Blue Yonder product team in its migration to Dask.
+ Coincidentally, we also talked with other attendees of the workshop whom were interested in such a functionality.
```

Comment:
> I wouldn't call the semaphore a result of the workshop since we were doing it anyhow. 
Technically right, but does the reader of this blog benefit from knowing this difference?

I would say, the workshop accelerated work on the semaphore significanlty and it is something that blocked LSF. We can add the angle about community requests, it fits in nice here.

Reply:
fixed typo